### PR TITLE
#47: Fixed wrong data type for double literals

### LINKED ISSUE #47: Some Double Literals Have Wrong Data Type
The Exasol dialect should always add a `CAST(... AS DOUBLE)` for literal_double. Otherwise some numbers like 1E0 get parsed as DECIMAL.

This is the follow-up to exasol/virtual-schema-common-java/issues/209.

How to reproduce:
```sql
SELECT * FROM vs.t WHERE a = CAST(1 AS DOUBLE);
```
The pushdown request for this query will contain a `literal_double`, but the generated pushdown will contain `WHERE a = 1E0` which gets parsed as DECIMAL.

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #47

### CODE REVIEW

File: doc/changes/changes_5.0.1.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+## Bugfixes:
+
+* #47: Fixed wrong data type for double literals
+
+## Dependency Updates
+
+### Compile Dependency Updates
+
+* Updated `com.exasol:virtual-schema-common-jdbc:9.0.1` to 9.0.2
```
Comment (morazow) [inline doc/changes/changes_5.0.1.md:17]:
```suggestion
* Updated `com.exasol:virtual-schema-common-jdbc:9.0.1` to `9.0.2`
```

Comment (morazow) [reply doc/changes/changes_5.0.1.md:17]: I think we put the single quotes on the next version also.

File: doc/changes/changes_5.0.1.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Exasol Virtual Schema Common 5.0.0, released 2021-03-XX
```
Comment (morazow) [inline doc/changes/changes_5.0.1.md:1]:
```suggestion
# Exasol Virtual Schema Common 5.0.1, released 2021-03-XX
```

File: src/test/java/com/exasol/adapter/dialects/exasol/AbstractExasolSqlDialectIT.java [hunk 3]
```diff
@@ -299,6 +299,22 @@ void testGeometryMapping() {
...
+        assertVsQuery("SELECT 10.2 FROM " + getVirtualTableName(this.virtualSchema, table),
+                table("DECIMAL").row(BigDecimal.valueOf(10.2)).matches());
+    }
+
+    @Test
+    void testDoubleLiteral() {
+        final Table table = createSingleColumnTable("BOOLEAN").insert(true);
+        this.virtualSchema = createVirtualSchema(this.sourceSchema);
+        assertVsQuery("SELECT CAST(10.2 as DOUBLE) FROM " + getVirtualTableName(this.virtualSchema, table),
+                table("DOUBLE PRECISION").row(10.2).matches());
```
Comment (morazow) [inline src/test/java/com/exasol/adapter/dialects/exasol/AbstractExasolSqlDialectIT.java:315]: So the idea is to cast to double when corresponding Exasol column is `double precision`?

**Review by morazow — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/adapter/dialects/exasol/AbstractExasolSqlDialectIT.java (hunk 3 above)
Comment (jakobbraun) [reply src/test/java/com/exasol/adapter/dialects/exasol/AbstractExasolSqlDialectIT.java:315]: yes. It's a workaround for Exasol to being not standard conform in SQL parsing.

File: doc/changes/changes_5.0.1.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+## Dependency Updates
+
+### Compile Dependency Updates
+
+* Updated `com.exasol:virtual-schema-common-jdbc:9.0.1` to 9.0.2
+
+### Test Dependency Updates
+
+* Updated `com.exasol:virtual-schema-common-jdbc:9.0.1` to 9.0.2
```
Comment (jakobbraun) [inline doc/changes/changes_5.0.1.md:21]:
```suggestion
* Updated `com.exasol:virtual-schema-common-jdbc:9.0.1` to `9.0.2`
```

File: doc/changes/changes_5.0.1.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+### Plugin Dependency Updates
+
+* Updated `com.exasol:project-keeper-maven-plugin:0.4.2` to 0.5.0
```
Comment (jakobbraun) [inline doc/changes/changes_5.0.1.md:25]:
```suggestion
* Updated `com.exasol:project-keeper-maven-plugin:0.4.2` to `0.5.0`
```

**Review by morazow — DISMISSED** [review]

File: doc/changes/changes_5.0.1.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# Exasol Virtual Schema Common 5.0.1, released 2021-03-XX
```
Comment (jakobbraun) [inline doc/changes/changes_5.0.1.md:1]:
```suggestion
# Exasol Virtual Schema Common 5.0.1, released 2021-03-18
```

**Review by morazow — APPROVED** [review]

**Pull request merged**